### PR TITLE
use a postsync hook job for deployment trigger

### PR DESCRIPTION
**Pull request by harshad16**
use a postsync hook job for deployment trigger
Signed-off-by: Harshad Reddy Nalla <hnalla@redhat.com>

## Description

https://github.com/AICoE/aicoe-cd/issues/233#issuecomment-767672303
we can test if resource hook can help us with triggering deploymentconfig.

### CODE REVIEW

File: user-api/overlays/test/deployment-trigger.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+metadata:
+  name: deployment-trigger
+  annotations:
+    argocd.argoproj.io/hook: PostSync
+    argocd.argoproj.io/hook-delete-policy: HookSucceeded
+spec:
+  template:
+    spec:
+      containers:
+        - name: chat-notification
```
Comment (goern) [inline user-api/overlays/test/deployment-trigger.yaml:13]: this should be changed for consistency

File: user-api/overlays/test/deployment-trigger.yaml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+          image: quay.io/openshift/origin-cli:latest
+          command:
+            - "oc"
```
Comment (goern) [inline user-api/overlays/test/deployment-trigger.yaml:16]: is this automatically authenticated as the default user of which namespace?!

**Review by goern — CHANGES_REQUESTED** [review]

File: user-api/overlays/test/deployment-trigger.yaml (hunk 2 above)
Comment (harshad16) [reply user-api/overlays/test/deployment-trigger.yaml:16]: not entirely sure about this, would like to test it out.

File: user-api/overlays/test/deployment-trigger.yaml (hunk 1 above)
Comment (harshad16) [reply user-api/overlays/test/deployment-trigger.yaml]: thanks

**Pull request merged**